src/Path/Include.hs

```diff
@@ -735,19 +740,41 @@ normalizeDir =
       normalizeRelDir p | p == relRootFP = ""
       normalizeRelDir p = p
 
+-- | Replaces consecutive path seps with single sep
+-- | and replaces alt sep with standard sep.
+normalizeAllSeps :: FilePath -> FilePath
+normalizeAllSeps = foldr norm []
+  where
+      norm ch [] = [ch]
+      norm ch path@(p0:_) |
+        FilePath.isPathSeparator ch && FilePath.isPathSeparator p0 = path
+      norm ch path |
+        FilePath.isPathSeparator ch = FilePath.pathSeparator:path
+      norm ch path = ch:path
+
+-- | Normalizes seps except at the beginning of path.
+normalizeSepsExceptLeading :: FilePath -> FilePath
+normalizeSepsExceptLeading path = normLeadingSeps ++ normalizeAllSeps rest
+  where leadingSeps = takeWhile FilePath.isPathSeparator path
+        normLeadingSeps = replicate (min 2 (length leadingSeps)) FilePath.pathSeparator
+        rest = dropWhile FilePath.isPathSeparator path
+
+-- | Normalizes seps only at the beginning of a path.
+normalizeLeadingSeps :: FilePath -> FilePath
+normalizeLeadingSeps (p0:p1:ps) |
+  FilePath.isPathSeparator p0 && FilePath.isPathSeparator p1 =
+    normalizeLeadingSeps (FilePath.pathSeparator:ps)
+normalizeLeadingSeps path = path
+
+-- | Normalizes seps only at the end of a path.
+normalizeTrailingSeps :: FilePath -> FilePath
+normalizeTrailingSeps = reverse . normalizeLeadingSeps . reverse
+
+-- | Applies file system-specific sep normalization following @FilePath.normalise@.
 normalizeFilePath :: FilePath -> FilePath
-#if MIN_VERSION_filepath(1,4,0)
-normalizeFilePath = FilePath.normalise
-#else
 normalizeFilePath
-  | IS_WINDOWS = FilePath.normalise
-  | otherwise = normalizeLeadingSeparators . FilePath.normalise
-    where
-        sep = FilePath.pathSeparator
-        normalizeLeadingSeparators (x1:x2:xs) | x1 == sep && x2 == sep
-            = normalizeLeadingSeparators (sep:xs)
-        normalizeLeadingSeparators x = x
-#endif
+  | IS_WINDOWS = normalizeSepsExceptLeading . FilePath.normalise
+  | otherwise  = normalizeLeadingSeps . FilePath.normalise
```

Comment:
I think we should try to get these extra normalizations upstreamed into `filepath`. As you already appear to be quite familiar with this topic, maybe you'd like to give this a try @rkoeninger?!

Reply:
You're suggesting I open an issue on https://github.com/haskell/filepath after this is merged? And if `filepath` gets released with these normalization functions, we'll come back and remove them from this library?

Reply:
> I can open an issue on https://github.com/haskell/filepath ?

That would be great! :)

> We'll come back and clear out the normalization functions once they're released in upstream.

I think we want to keep compatibility with the `filepath` versions from the last ~3 years. But eventually we should be able to delete most of our normalization code.